### PR TITLE
Lzk/public eks

### PR DESCRIPTION
**Pull request by lzianekhodja**

### CODE REVIEW

File: infrastructure/quick-deploy/aws/vpc/variables.tf [hunk 1]
```diff
@@ -54,22 +54,11 @@ variable "vpc" {
       peer_vpc_ids = list(string)
     })
   })
-  default = {
```
Comment (aneojgurhem) [inline infrastructure/quick-deploy/aws/vpc/variables.tf:57]: the default value could stay here

File: infrastructure/quick-deploy/aws/eks/variables.tf [hunk 2]
```diff
@@ -104,4 +104,11 @@ variable "eks_operational_worker_groups" {
 variable "eks_worker_groups" {
   description = "List of EKS worker node groups"
   type        = any
+}
+
+# Enable EKS public access
+variable "enable_public_eks_access" {
+  description = "Enable EKS public access"
+  type        = string
```
Comment (aneojgurhem) [inline infrastructure/quick-deploy/aws/eks/variables.tf:112]: Is there a boolean type ?

**Comment by ngruelaneo** [comment]
Which version of Armonik is concerned?

File: infrastructure/quick-deploy/aws/vpc/Makefile [hunk 3]
```diff
@@ -9,6 +9,7 @@ export SUFFIX?=main
 export REGION?=eu-west-3
 export PROFILE?=default
 export TFSTATE_BUCKET_NAME?=armonik-tfstate-$(SUFFIX)
+export PUBLIC_VPC=?true
```
Comment (lemaitre-aneo) [inline infrastructure/quick-deploy/aws/vpc/Makefile:12]:
```suggestion
export PUBLIC_VPC?=true
```

File: infrastructure/quick-deploy/aws/eks/Makefile [hunk 4]
```diff
@@ -8,8 +8,9 @@ export TF_DATA_DIR?=$(GENERATED_DIR)
 export SUFFIX?=main
 export REGION?=eu-west-3
 export PROFILE?=default
-export TFSTATE_BUCKET_NAME?=armonik-tfstate
+export TFSTATE_BUCKET_NAME?=armonik-tfstate-$(SUFFIX)
 export VPC_PARAMETERS_FILE?=$(CURRENT_DIR)/../vpc/generated/vpc-output.json
+export PUBLIC_ACCESS_EKS=?true
```
Comment (lemaitre-aneo) [inline infrastructure/quick-deploy/aws/eks/Makefile:13]:
```suggestion
export PUBLIC_ACCESS_EKS?=true
```

File: infrastructure/quick-deploy/aws/eks/locals.tf [hunk 5]
```diff
@@ -9,10 +9,11 @@ resource "random_string" "random_resources" {
...
 locals {
-  random_string = random_string.random_resources.result
-  suffix        = var.suffix != null && var.suffix != "" ? var.suffix : local.random_string
-  cluster_name  = try(var.vpc.eks_cluster_name, "armonik-eks-${local.suffix}")
-  kms_name      = "armonik-kms-eks-${local.suffix}-${local.random_string}"
+  random_string                  = random_string.random_resources.result
+  suffix                         = var.suffix != null && var.suffix != "" ? var.suffix : local.random_string
+  cluster_name                   = try(var.vpc.eks_cluster_name, "armonik-eks-${local.suffix}")
+  kms_name                       = "armonik-kms-eks-${local.suffix}-${local.random_string}"
+  cluster_endpoint_public_access = try(tobool(try(lower(var.enable_public_eks_access), false)), false)
```
Comment (lemaitre-aneo) [inline infrastructure/quick-deploy/aws/eks/locals.tf:16]: This will be much simpler once `var.enable_public_eks_access` is a `bool`

**Review by ddubuc — APPROVED** [review]

File: infrastructure/quick-deploy/aws/vpc/variables.tf [hunk 6]
```diff
@@ -54,22 +54,11 @@ variable "vpc" {
...
-      enabled      = false
-      peer_vpc_ids = []
-    }
-  }
+}
+
+# Enable public VPC
+variable "enable_public_vpc" {
+  description = "Enable public VPC"
+  type        = string
```
Comment (lemaitre-aneo) [inline infrastructure/quick-deploy/aws/vpc/variables.tf:62]: Should qlso be a `bool`

**Review by lemaitre-aneo — APPROVED** [review]

**Pull request merged**